src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -1163,6 +1163,15 @@ class VariantSampleMatrix[RPK, RK, T >: Null](val hc: HailContext, val metadata:
     vaSignature.insert(sig, path)
   }
 
+  def insertIntoRow[PC](typeToInsert: Type, path: List[String], makePartitionContext: () => PC,
+    inserter: (PC, RegionValue, RegionValueBuilder) => Unit): VariantSampleMatrix[RPK, RK, T] = {
```

Comment:
I suspect the type inference will work better if you place `inserter` in a separate argument list. I suspect the `() => PC` will rarely need a type annotation, and the Scala inferencer will be able to deduce the type of the lambda without an explicit instantiation of `PC`.